### ISSUE TITLE
Aws-nuke fails on Deleting EBS Volumes

### ISSUE THREAD

**Issue by josh-padnick** [gruntwork-io/aws-nuke#10]
While doing a fresh `aws-nuke` run, I received the following error:

```
INFO[2018-02-18T12:07:48-07:00] No Auto Scaling Groups to nuke in region eu-central-1
INFO[2018-02-18T12:07:48-07:00] No Elastic Load Balancers to nuke in region eu-central-1
INFO[2018-02-18T12:07:48-07:00] No V2 Elastic Load Balancers to nuke in region eu-central-1
INFO[2018-02-18T12:07:48-07:00] Terminating all EC2 instances in region eu-central-1
INFO[2018-02-18T12:07:49-07:00] Terminated EC2 Instance: i-0c61c9e0e5878f441
	INFO[2018-02-18T12:08:54-07:00] [OK] 1 instance(s) terminated in eu-central-1
INFO[2018-02-18T12:08:54-07:00] Deleting all EBS volumes in region eu-central-1
ERRO[2018-02-18T12:08:55-07:00] [Failed] VolumeInUse: Volume vol-0b43429069414b331 is currently attached to i-05333ad8fe9a89158
	status code: 400, request id: eeea65fe-4c56-414d-89b0-1878293db105
```

This makes me realize that part of deleting an EBS Volume means repeatedly checking that it has in fact been detached before initiating its termination.

**Comment by josh-padnick** [comment]
Also, re-running the tool still doesn't fix this issue, so I guess it's assuming that only EBS Volumes that have been manually detached should be deleted? That seems like an unreasonable expectation of a user who wants to "nuke" their account.

**Comment by josh-padnick** [comment]
This also makes me realize that you should only attempt to detach EBS Volumes from EC2 Instances that do not have EC2 Instance Termination protection.

**Comment by josh-padnick** [comment]
Sorry for the repeated messages here, but, unfortunately `aws-nuke` is not really usable in its current state :( since repeated runs yield the same issue:

```
Are you sure you want to nuke all listed resources? Enter 'nuke' to confirm: nuke
INFO[2018-02-18T12:15:06-07:00] Deleting all Auto Scaling Groups in region eu-west-1
INFO[2018-02-18T12:15:07-07:00] Deleted Auto Scaling Group: jenkins-example-0
INFO[2018-02-18T12:15:56-07:00] [OK] 1 Auto Scaling Group(s) deleted in eu-west-1
INFO[2018-02-18T12:15:56-07:00] No Elastic Load Balancers to nuke in region eu-west-1
INFO[2018-02-18T12:15:56-07:00] Deleting all V2 Elastic Load Balancers in region eu-west-1
INFO[2018-02-18T12:15:57-07:00] Deleted ELBv2: arn:aws:elasticloadbalancing:eu-west-1:087285199408:loadbalancer/app/aws-nuke-test-7SSdhR/36b9b62a0b6c8614
INFO[2018-02-18T12:15:57-07:00] Deleted ELBv2: arn:aws:elasticloadbalancing:eu-west-1:087285199408:loadbalancer/app/jenkins-example-alb/4f731144cf954cb7
INFO[2018-02-18T12:15:57-07:00] [OK] 2 V2 Elastic Load Balancer(s) deleted in eu-west-1
INFO[2018-02-18T12:15:57-07:00] No EC2 instances to nuke in region eu-west-1
INFO[2018-02-18T12:15:57-07:00] Deleting all EBS volumes in region eu-west-1
INFO[2018-02-18T12:15:58-07:00] Deleted EBS Volume: vol-09cf67a1f0df7c468
INFO[2018-02-18T12:15:59-07:00] Deleted EBS Volume: vol-09ae01d07e21d39e6
ERRO[2018-02-18T12:15:59-07:00] [Failed] VolumeInUse: Volume vol-02281951cea62796b is currently attached to i-0061e3d23e018c6e0
	status code: 400, request id: 15d907c8-95e6-4c7d-a31b-aad92a07d6b4
```

**Comment by tonerdo** [comment]
Thanks for reporting this @josh-padnick. #11 fixes this issue

**Comment by tonerdo** [comment]
Fixed in #11

**Comment by tonerdo** [comment]
Fixed in #11
